back-end/java/main/java/src/DormRoom/DormRoom.java

```diff
@@ -0,0 +1,157 @@
+package src.DormRoom;
+
+/**
+ * The {@code DormRoom} class represents a dormitory room with various attributes such as size,
+ * capacity, building location, and amenities. It is designed to encapsulate data about a single
+ * dorm room, including its building, the type of bathroom it has, and whether it includes a kitchen
+ * or is part of a suite. Instances of this class are immutable once constructed.
+ *
+ * <p>Usage example:
+ * <pre>
+ * DormRoom myRoom = new DormRoom(200, 255, 2,
+ *    'example.com', true, false,
+ *    BathroomType.Private, "Barbour Hall");
+ * </pre>
+ */
+public class DormRoom {
+
+    /**
+     * The square footage of the room.
+     */
+    final int roomSize;
+
+    /**
+     * The room number, as assigned by the Office of Residential Life. For example, "Room 255".
+     */
+    final int roomNumber;
```

Comment:
Room number may not be an int. Some rooms are tagged with letters as well (e.g. BUXTON 316A). Additionally, room numbers in both the portal and the CSV they get are of the form <building> <number> (e.g. BUXTON 316A) for normal rooms and <building> <suite num> <room num> (e.g. BARBOUR 080 084) for suites. Representing this as a string will make it easier for users to identify what room on the portal it corresponds to.

Reply:
This will also auto-correct rooms like 040 to just 40, which is workable but annoying. We shouldn't be sorting or filtering by room number anyway so it shouldn't hurt changing to string